firefox-ios/Storage/Rust/RustRemoteTabs.swift

```diff
@@ -30,19 +31,20 @@ public class RustRemoteTabs {
         self.logger = logger
 
         queue = DispatchQueue(label: "RustRemoteTabs queue: \(databasePath)", attributes: [])
+        commandQueue = DispatchQueue(label: "RustRemoteTabs command queue", attributes: [])
```

Comment:
```suggestion
```

Oh, I missed that we already had a serial dispatch queue for `RustRemoteTabs`! Let's reuse `queue`, instead of making a second `commandQueue`.

Most times, the queue is going to be sitting idle, and we already use it for synchronizing state (the `queue.sync` calls in `reopenIfClosed` and `forceClose`) and I/O work (`setLocalTabs`, `getAll`, etc). Reusing the queue will avoid potentially asking GCD to spin up another thread when we have commands to send.

Second, since `RemoteTabsCommandQueue` shares the same state (behind the same mutex) as `TabsStore` in Rust, I don't think a separate queue would give us any additional concurrency. IOW, even if, say, `getAll()` and `addRemoteCommand()` were called concurrently, and ran on two different threads, they both need to take the same mutex that protects the SQLite database connection in Rust, so one will always need to wait on the other.